grunt_tasks/plugin.js

```diff
@@ -68,105 +78,126 @@ module.exports = function (grunt) {
         grunt.config.merge(cfg);
     };
 
-    grunt.config.merge({
-        default: {
-            plugin: {}
+    var configurePluginForBuilding = function (dir) {
+        var plugin = path.basename(dir);
+        var json = path.resolve(dir, 'plugin.json');
+        var yml = path.resolve(dir, 'plugin.yml');
+        var config = {}, npm;
+
+        if (!fs.lstatSync(dir).isDirectory()) {
+            grunt.fail.warn('Plugin directory not found: ' + dir);
+            return;
         }
-    });
 
-    /**
-     * Insert individual plugins into the plugin meta task.
-     */
-    grunt.file.expand(grunt.config.get('pluginDir') + '/*')
-        .forEach(function (dir) {
-            var plugin = path.basename(dir);
-            var json = path.resolve(dir, 'plugin.json');
-            var yml = path.resolve(dir, 'plugin.yml');
-            var config = {}, npm;
+        if (fs.existsSync(json)) {
+            config = grunt.file.readYAML(json);
+        }
+        if (fs.existsSync(yml)) {
```

Comment:
I'd say the ordering is fine, but I'd use an `else if` and log a message indicating which is being used. We should also not the order preferences in the documentation.